assets/javascripts/discourse/components/chat-channel-selector-modal-inner.js

```diff
@@ -140,7 +141,7 @@ export default Component.extend({
             }
           });
           this.setProperties({
-            channels: channels.map((c) => EmberObject.create(c)),
+            channels: channels.map((channel) => ChatChannel.create(channel)),
```

Comment:
This is a bigger topic, but… we should not manually create ChatChannel instances all over the place. The same with calling ajax in components like here.

We should extract that logic to model/adapter/service, and then rely on the store, e.g. `this.store.find("chat-channel", id)` or `this.store.findFiltered("chat-channel", { filter })`